OktaIdxExample/OktaIdxExampleUITests/SelfServiceRegistrationScenarioTests.swift

```diff
@@ -276,23 +277,31 @@ class SelfServiceRegistrationScenarioTests: XCTestCase {
         let codeExpectation = expectation(description: "Email code received.")
         var emailCode: String?
         
-        a18nProfile.message { (message: A18NProfile.EmailMessage?, error) in
-            emailCode = message?.content
+        // Wait the code received by email box
```

Comment:
Arbitrary sleep statements bother me, from a timing perspective. Why don't you use `waitForCode` instead, defined in the `CodeReceiver` protocol?  You can also run `reset` on the receiver at the beginning of the test (in `setup` perhaps) to ensure any previous messages are deleted.

The `waitForCode` method will utilize a polling interval to periodically check until the email comes in.

Reply:
Good advice.